### ISSUE TITLE
Remove newlines after `def`

### ISSUE THREAD

**Issue by saroad2** [psf/black#3033]
**TL;DR**

There shouldn't be a newline right after `def` statement.

**Describe the style change**

Newlines are great. They help us organize our code and leave room for the coders and reviewers to breath.
However, too many newlines can make simple code gigantic and massive. This is unnecessary in a lot of cases.

I want to argue that having a newline right after `def` is redundant and *Black* should remove it

**Examples in the current _Black_ style**

Right now *Black* won't change the following code:

```python
def f():

    print("No newline above me!")

    print("There is a newline above me, and that's OK!")
```

**Desired style**

*Black* should change the code above to:

```python
def f():
    print("No newline above me!")

    print("There is a newline above me, and that's OK!")
```

**What about classes?**

I think that the issue above is not relevant to classes.

A new line after a `class` definition makes sense to me, since there is no actual meaning for the methods definitions order.

The newline after `class` definition makes room for the reader to breath before diving in to each method of the class.

So, *Black* should not change the following code:
```python
class MyClass:

    def the_newline_above_me_should_stay(self):
        print("Leave the newline alone!")
```

**Comment by JelleZijlstra** [comment]
Duplicate of #902.

For what it's worth I think this is a reasonable idea; feel free to submit a PR.
